src/main/groovy/com/bmuschko/gradle/docker/internal/SystemConfig.groovy

```diff
@@ -0,0 +1,50 @@
+package com.bmuschko.gradle.docker.internal
+
+import com.google.common.annotations.VisibleForTesting
+import groovy.transform.CompileStatic
+
+/**
+ * The internal abstraction on top of JVM class {@link System}. The main purpose is to provide a way to manipulate
+ * these properties and environments in the unit test environment.
+ */
+@CompileStatic
+class SystemConfig {
+
+    private static Map<String, String> OVERRIDE_PROPERTIES = [:]
```

Comment:
I think this internal "cache" makes things unnecessarily brittle. I don't think there's a big impact to querying the info multiple times. Please also hidden this behind an interface and make it non-static.

Reply:
Removed `OVERRIDE_PROPERTIES`, but still keep `OVERRIDE_ENVS`, as there is no similar functionality for system environments. I think `SystemConfig` can be an interface with default implementation which use System class to get system environments. For unit tests, I can pass my own implementation which can override some envs. What do you think?